### PR TITLE
package: add desktop entry

### PR DESCRIPTION
**Pull request by TheOnAndOnlyZenomat**
This adds a desktop entry that shows up in application launchers

### CODE REVIEW

**Comment by TheOnAndOnlyZenomat** [comment]
Ok, the non free editions, at least I think, I can only test with the personal license, include a `docs` dir, which also includes the icon I used. Apparently that is not the case with the free edition. Is there a better way than including the logo in this repository and copying it over? Would it be ok with you, if I included the logo in the repo? @jchv

File: package.nix [hunk 1]
```diff
@@ -60,12 +63,26 @@ stdenv.mkDerivation {
     "wayland"
   ];
   buildPhase = ":";
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Binary Ninja";
+      exec = "binaryninja";
+      icon = "logo";
```
Comment (jchv) [inline package.nix]: I don't think the icon for binaryninja should be called `logo` in the global namespace of installed XDG icons. Let's rename the file.
```suggestion
      icon = "binaryninja";
```

File: package.nix [hunk 2]
```diff
@@ -60,12 +63,26 @@ stdenv.mkDerivation {
...
+  ];
+
   installPhase = ''
     runHook preInstall
 
     mkdir -p $out/bin
     mkdir -p $out/opt
+    mkdir -p $out/share/pixmaps
     cp -r * $out/opt
+    cp $out/opt/docs/img/logo.png $out/share/pixmaps/
```
Comment (jchv) [inline package.nix]: For license reasons it'd be preferable, IMO, to not include the actual logo in this repo. However, that's OK: the `/docs/` folder in the Binary Ninja zip files is just the online documentation. This same image is also available on the online docs site, so we can do this:
```suggestion
    cp ${
      fetchurl {
        url = "https://docs.binary.ninja/img/logo.png";
        hash = "sha256-TzGAAefTknnOBj70IHe64D6VwRKqIDpL4+o9kTw0Mn4=";
      }
    } $out/share/pixmaps/binaryninja.png
```
(Note: of course, we need to add `fetchurl` to the inputs at the top. My preference would be after `python3` and before the separated parameter inputs below it.)

File: package.nix (hunk 1 above)
Comment (TheOnAndOnlyZenomat) [reply package.nix]: That makes a lot of sense

File: package.nix (hunk 2 above)
Comment (TheOnAndOnlyZenomat) [reply package.nix]: I was hoping such a solution would exists, I just couldn't figure that out with my nix knowledge. That looks perfect

**Comment by TheOnAndOnlyZenomat** [comment]
I'll try to get a fix in, when I'm back home later today

**Comment by TheOnAndOnlyZenomat** [comment]
Ok, I think everything looks good now. Feel free to suggest any changes

**Review by jchv — APPROVED** [review]

**Comment by TheOnAndOnlyZenomat** [comment]
Perfect, thank you

**Pull request merged**